Add retroactive pfile deidentifyer oneoff script

ambrussimon opened the pull request:
Supersedes the [original PR](https://github.com/flywheel-io/team/pull/7) opened in the team repo.

* Rewritten as a core one-off for direct file system/mongo access + ops simplicity
* Support efile/pfile/pfile header (can select one or more)
* Searching in acquisitions _and_ analyses
* Writes list of offending files at the end
* Added some error handling

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

nagem commented:
Are all new files entering the system de-identified? The script doesn't have to worry about "bad" p and e files entering the system while it's running, correct?

nagem commented:
Do we expect any of these files to get replaced by normal use during the de-id process?

nagem commented:
If the answer to my two questions above are no, I'd say this looks good from my perspective. I don't have a lot of insight into how the de-indentification process is supposed to work, but the structure around getting and editing the files looks good. LGTM.

ambrussimon commented:
Rebased + added handling/logging for
* Invalid files (cannot open or de-id) => log error, skip
* Changed files (cannot find orig hash in mongo) => log warning, no mongo update for that container

The script is not handling files (good or bad) entering the system while it's running. Newest reaper should be uploading everything properly de-identified. I hope we can ignore manual uploads of pfiles (& co) containing PHI.

nagem commented:
Changes look good 👍

ambrussimon commented:
Bump @gsfr. Let me know what time might work for everyone next week.

ryansanford commented on bin/oneoffs/retro_deid.py:
```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python
+"""
+Retroactively de-identify EFiles, PFiles and PFile headers.
```
Can we add a little more meat to this description? Are both files and DB modified? Just files? Are files skipped on error, or does script halt?

ryansanford commented on bin/oneoffs/retro_deid.py:
```diff
@@ -0,0 +1,294 @@
+Retroactively de-identify EFiles, PFiles and PFile headers.
+
+* Should be run with the same UID as Core (adds de-identified files to storage).
+* Requires DB version 43 or above (analysis inputs under inputs key).
```
Can the script check for this, and exit with warning if this isn't met?

ryansanford commented on bin/oneoffs/retro_deid.py:
```diff
@@ -0,0 +1,294 @@
+* Requires DB version 43 or above (analysis inputs under inputs key).
+* Scanning includes containers marked as 'deleted'.
+* Does not remove the (unreferenced) PHI files from storage.
+* Incompatible with the Object Storage branch (assumes local storage).
```
Can the script check for this, and exit with warning if this isn't met?

gsfr commented:
@ryansanford Are you good with these updates? We'd like to get this merged before we run it on Wednesday. I need to inform the customer of our plans.

gsfr commented:
@ambrussimon I just realized that all/most p-files in prod systems are gzipped. Doesn't look like we are accounting for that here.

ambrussimon commented:
@gsfr Thanks for the catch. Based on the reaper implementation (past and present) known to me, I assumed _all_ pfiles are gzipped, and fixed/tested (now using proper reaper upload) accordingly.

Please confirm that no uncompressed pfiles are to be expected.

gsfr commented:
Yes, I think it is safe to assume no uncompressed p-files.
